File: C++/Medium/Accepted/most-stones-removed-with-same-row-or-column_2.cpp
class Solution {
public:
    int removeStones(vector<vector<int>>& stones) {
        
        int n = stones.size();
        vector<int> parent(n);
        for (int i = 0; i < n; ++i) {
            parent[i] = i;
        }
        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                if (stones[i][0] == stones[j][0] || stones[i][1] == stones[j][1]) {
                    parent[find(parent, i)] = find(parent, j);
                }
            }
        }
        unordered_set<int> seen;
        for (int i = 0; i < n; ++i) {
            seen.insert(find(parent, i));
        }
        return n - seen.size();
    }
    int find(vector<int>& parent, int index) {
        if (parent[index] != index) {
            parent[index] = find(parent, parent[index]);
        }
        return parent[index];
    }
};

File: Rust/Medium/CompileError/most-stones-removed-with-same-row-or-column_2.rs
impl Solution {
    pub fn remove_stones(stones: Vec<Vec<i32>>) -> i32 {
        
        let mut ans = 0;
        let mut uf = UnionFind::new(20001);
        for stone in stones {
            uf.union(stone[0] as usize, stone[1] as usize + 10000);
        }
        for stone in stones {
            if uf.find(stone[0] as usize) == uf.find(stone[1] as usize + 10000) {
                ans += 1;
            }
        }
        (stones.len() - ans) as i32
    }
}